src/SlackClientFactory.ts

```diff
@@ -220,6 +225,9 @@ export class SlackClientFactory {
             const auth = (await slackClient.auth.test()) as AuthTestResponse;
             const user = (await slackClient.users.info({user: auth.user_id})) as UsersInfoResponse;
             log.debug("Created new team client for", teamInfo.team.name);
+            if (this.updatePuppetCount) {
+                this.updatePuppetCount(teamInfo.team.id, 1);
+            }
```

Comment:
We're on Typescript 3.8, which means we can use [optional chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining)
```suggestion
            this.updatePuppetCount?.(teamInfo.team.id, 1);
```